src/mesh/coordinates.cxx

```diff
@@ -505,6 +505,43 @@ int Coordinates::jacobian() {
   return 0;
 }
 
+/** This function calculates the L2 vector norm of the Linf matrix norm of gij*g_ij - I.
+ * 	This function can be used to test the consistency of metric the co- and contravariant tensors
+ **/
+BoutReal Coordinates::gijXg_ijMinusI(){
+	BoutReal globalError = 0.;
+	BoutReal localError = 0.;
+	///2d array of field3d pointers. Using array because vector of const references aor const pointers is not allowed
+	Field2D * const gij[3][3] = {{&g11,&g12,&g13}, {&g12,&g22,&g23}, {&g13,&g23,&g33}};
+	Field2D * const g_ij[3][3]= {{&g_11,&g_12,&g_13}, {&g_12,&g_22,&g_23}, {&g_13,&g_23,&g_33}};
+
+	BoutReal res[3][3] = {{0.,0.,0.},{0.,0.,0.},{0.,0.,0.}};
+	const BoutReal I[3][3] = {{1.,0.,0.},{0.,1.,0.},{0.,1.,0.}};
+
+	for(int x=mesh->xstart;x<mesh->xend;x++)for(int y=mesh->ystart;y<mesh->yend;y++)
+	{//do matrix multiplication for each point in space
+		for(int i = 0; i<3;i++)
+		{
+			for(int j = 0; j<3;j++)
+			{
+				for(int k = 0; k<3;k++){
+
+					res[i][j] += (*gij[i][k])(x,y) * (*g_ij[k][j])(x,y);
+				}
+				res[i][j] -=  I[i][j];
```

Comment:
yes. I am an idiot. Sorry. I did it in a hurry. 

I cannot test my revised version though because I cannot run any program due to problems with netcdf....